src/wallet/wallet.cpp

```diff
@@ -333,26 +338,46 @@ bool CWallet::ChangeWalletPassphrase(const SecureString& strOldWalletPassphrase,
         CKeyingMaterial vMasterKey;
         BOOST_FOREACH(MasterKeyMap::value_type& pMasterKey, mapMasterKeys)
         {
-            if(!crypter.SetKeyFromPassphrase(strOldWalletPassphrase, pMasterKey.second.vchSalt, pMasterKey.second.nDeriveIterations, pMasterKey.second.nDerivationMethod))
+            if(!crypter.SetKeyFromPassphrase(
+                    strOldWalletPassphrase,
+                    pMasterKey.second.vchSalt,
+                    pMasterKey.second.nDeriveIterations,
+                    pMasterKey.second.nDerivationMethod,
+                    pMasterKey.second.vchOtherDerivationParameters))
                 return false;
             if (!crypter.Decrypt(pMasterKey.second.vchCryptedKey, vMasterKey))
                 return false;
             if (CCryptoKeyStore::Unlock(vMasterKey))
             {
                 int64_t nStartTime = GetTimeMillis();
-                crypter.SetKeyFromPassphrase(strNewWalletPassphrase, pMasterKey.second.vchSalt, pMasterKey.second.nDeriveIterations, pMasterKey.second.nDerivationMethod);
+                crypter.SetKeyFromPassphrase(
+                        strNewWalletPassphrase,
+                        pMasterKey.second.vchSalt,
+                        pMasterKey.second.nDeriveIterations,
+                        pMasterKey.second.nDerivationMethod,
+                        pMasterKey.second.vchOtherDerivationParameters);
                 pMasterKey.second.nDeriveIterations = pMasterKey.second.nDeriveIterations * (100 / ((double)(GetTimeMillis() - nStartTime)));
 
                 nStartTime = GetTimeMillis();
-                crypter.SetKeyFromPassphrase(strNewWalletPassphrase, pMasterKey.second.vchSalt, pMasterKey.second.nDeriveIterations, pMasterKey.second.nDerivationMethod);
+                crypter.SetKeyFromPassphrase(
+                        strNewWalletPassphrase,
+                        pMasterKey.second.vchSalt,
+                        pMasterKey.second.nDeriveIterations,
+                        pMasterKey.second.nDerivationMethod,
+                        pMasterKey.second.vchOtherDerivationParameters);
                 pMasterKey.second.nDeriveIterations = (pMasterKey.second.nDeriveIterations + pMasterKey.second.nDeriveIterations * 100 / ((double)(GetTimeMillis() - nStartTime))) / 2;
```

Comment:
As you can see from the fact there's no diff on this line, it's just the formula from the old derivation method. But you can see that the goal is to scale the number of iterations such that there are fewer on a slower machine. I doubt it is optimal for Argon2i however, given that the iteration numbers are much lower, so we should decide on either an alternate formula or an alternate strategy.